prime-dotnet-webapi/Services/Clients/AddressAutocomplete/AddressAutocompleteClient.cs

```diff
@@ -0,0 +1,186 @@
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Prime.Services.Clients
+{
+    public class AddressAutocompleteClient : IAddressAutocompleteClient
+    {
+        private readonly HttpClient _client;
+        private readonly ILogger _logger;
+
+        public AddressAutocompleteClient(
+            HttpClient client,
+            ILogger<AddressAutocompleteClient> logger)
+        {
+            // Auth header and api-key are injected in Startup.cs
+            _client = client;
+            _logger = logger;
+        }
+
+        public async Task<IEnumerable<AddressAutocompleteFindResponse>> Find(string searchTerm, string lastId)
+        {
+            //Build the url
+            var url = "Find/v2.10/json3ex.ws?";
+            url += "Key=" + System.Web.HttpUtility.UrlEncode(PrimeConstants.ADDRESS_VALIDATION_API_KEY);
```

Comment:
@james-hollinger How do i set a get variable key in the startup httpclient setup?

Reply:
You could dependancy inject a singleton object holding the key, something like

services.AddSingleton(new AddressValidationCredentials { Key = PrimeConstants.ADDRESS_VALIDATION_API_KEY });

in Startup.cs, and then you could change the constructor to
public AddressAutocompleteClient( HttpClient client, ILogger<AddressAutocompleteClient> logger, AddressValidationCredentials credentials)